File: compiled/julia/cdmsbats/DataRecord.jl
include("/home/tdoughty1/Workspace/programming/julia/cdmsbats/CDMSRawFileStream.jl")

abstract DataRecord

function ReadRecord(recPtr::DataRecord, filePtr::CDMSRawFileStream, RecordLength::Int, mode::Int, debug::Bool)

    Record = ReadWords(filePtr, RecordLength, mode)

    StoreValues(Record)

    if(debug)
            PrintValues(recPtr)
    end
end

function ReadRecord(recPtr::DataRecord, filePtr::CDMSRawFileStream, RecordLength::Int, mode::Int)

    Record = ReadWords(filePtr, RecordLength, mode)

    StoreValues(Record)
end

File: compiled/cpp/cdmsbats/datarecord.cpp
#include <string>

#include "datarecord.h"

void DataRecord::ReadRecord(CDMSRawFileStream* filePtr, int RecordLength, uint32_t mode, bool debug)
{
    uint32_t *Record;
    Record = new uint32_t[RecordLength/4];

    filePtr->ReadWords(RecordLength, Record);

    StoreValues(Record);

    if(debug)
    {
        PrintValues();
    }
}

void DataRecord::ReadRecord(CDMSRawFileStream* filePtr, int RecordLength, int32_t mode, bool debug)
{
    int32_t *Record;
    Record = new int32_t[RecordLength/4];

    filePtr->ReadWords(RecordLength, Record);

    StoreValues(Record);

    if(debug)
    {
        PrintValues();
    }
}
